src/main/groovy/org/transmartproject/batch/highdim/beans/AbstractStandardHighDimJobConfiguration.groovy

```diff
@@ -411,7 +105,10 @@ abstract class AbstractStandardHighDimJobConfiguration extends AbstractJobConfig
         step
     }
 
-    abstract ItemWriter<? extends DataPoint> dataPointWriter()
+    @Bean
```

Comment:
Should be at least job scoped (possible step scoped, if you expect it could be used more than once in the same job), since it has job-specific state in the variable `samplePatientMap`.